File: src/net/dinkla/raytracer/math/Point2DF.scala
package net.dinkla.raytracer.math

/**
* Created by Dinkla on 10.07.2015.
*/

final class Point2DF(x2: Float, y2: Float) extends Point2D[Float](x2, y2) {

   override val x: Float = x2
   override val y: Float = y2

   override def length: Float = math.sqrt(sqrLength).toFloat

   override def unary_-(): Point2D[Float] = new Point2DF(-x, -y)

   override def plus(v: Vector2D[Float]): Point2D[Float] = new Point2DF(x + v.x, y + v.y)

   override def plus(p: Point2D[Float]): Vector2D[Float] = new Vector2DF(x + p.x, y + p.y)

   override def toVector: Vector2D[Float] = new Vector2DF(x, y)

}

File: src/net/dinkla/raytracer/math/Point2DF.java
package net.dinkla.raytracer.math;

/**
 * Created by IntelliJ IDEA.
 * User: jorndinkla
 * Date: 10.04.2010
 * Time: 15:17:43
 * To change this template use File | Settings | File Templates.
 */
public class Point2DF extends Element2D {

    public Point2DF(float x, float y) {
        super(x, y);
    }

    public Vector2DF plus(final Point2DF v) {
        return new Vector2DF(x + v.x, y + v.y);
    }

    public Point2DF plus(final Vector2DF v) {
        return new Point2DF(x + v.x, y + v.y);
    }

    public Vector2DF minus(final Point2DF v) {
        return new Vector2DF(x - v.x, y - v.y);
    }

    public Point2DF minus(final Vector2DF v) {
        return new Point2DF(x - v.x, y - v.y);
    }

}